### ISSUE TITLE
Template requires caluse causes false error report from indirection.

### ISSUE THREAD

**Issue by InKryption** [microsoft/vscode-cpptools#7435]
Issue Type: <b>Bug</b>

All the following code compiles prefectly fine on "gcc -Wall -Wextra -Wpedantic -pedantic -pedantic-errors",

Compiles and parses fine:
template<typename T>
decltype(auto) test1(T&& v) {
	return -v.get_data();
}

Compiles fine but fails to parse:
template<typename T>
concept MINUS = requires(T t) { {-t.get_data()} };

template<MINUS T> /* The same occurs with an in-place requires expression */
decltype(auto) test_TMP(T&& v) {
	return -v.get_data();
}

I've been skulking at it for a few hours now, and I can't pin-point "exact" culprits, but it definitely seems to have something to do with applying the unary minus operator to the result of a method in a requires expression.

Extension version: 1.3.1
VS Code version: Code 1.55.2 (3c4e3df9e89829dce27b7b5c24508306b151f30d, 2021-04-13T09:35:57.887Z)
OS version: Windows_NT x64 10.0.19042
Remote OS version: Linux x64 4.19.128-microsoft-standard

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD A9-9420e RADEON R5, 5 COMPUTE CORES 2C+3G   (2 x 1797)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: disabled_off<br>oop_rasterization: enabled<br>opengl: enabled_on<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|11.46GB (6.67GB free)|
|Process Argv|--folder-uri=vscode-remote://wsl+Debian/home/ink/dev/CXX/GenericVecMath --remote=wsl+Debian --crash-reporter-id 057ea1d6-98a1-4c45-ba88-dd945febf03d|
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|WSL: Debian|
|OS|Linux x64 4.19.128-microsoft-standard|
|CPUs|AMD A9-9420e RADEON R5, 5 COMPUTE CORES 2C+3G (2 x 1796)|
|Memory (System)|8.91GB (8.20GB free)|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspyt653:30270858
pythonvspyt602cf:30294773
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstry244:30276681
pythonvspyt639:30291489
pythontb:30283811
vspre833:30267464
pythonptprofiler:30281270
vshan820:30294714
pythondataviewer:30285071
vscus158:30286553
vscgsv2:30294352

```

</details>

**Comment by sean-mcmanus** [comment]
The code doesn't compile for me with gcc 10.3 -- what version of gcc are you using? You're using -std=c++20, right? Is this a bug with IntelliSense showing errors it shouldn't? What does "compiles file but fails to parse" mean? Usually parse and compile mean the same thing.

I get errors:
```
<source>:2:22: error: 'test1' is not a template function
    2 | decltype(auto) test1(T&& v) {
      |                      ^
<source>:2:21: error: expected ';' before '(' token
    2 | decltype(auto) test1(T&& v) {
      |                     ^
      |                     ;
<source>:8:1: error: C++20 concept definition syntax is 'concept <name> = <expr>'
    8 | concept MINUS = requires(T t) { {-t.get_data()} };
      | ^~~~~~~
<source>:8:1: error: a non-template variable cannot be 'concept'
<source>:8:14: error: expected ';' before '=' token
    8 | concept MINUS = requires(T t) { {-t.get_data()} };
      |              ^~
      |              ;
<source>:11:25: error: 'test_TMP' is not a template function
   11 | decltype(auto) test_TMP(T&& v) {
      |                         ^
<source>:11:24: error: expected ';' before '(' token
   11 | decltype(auto) test_TMP(T&& v) {
      |                        ^
```

**Comment by InKryption** [comment]
Oh Christ, I must have been completely braindead when I posted this issue. Yeah, that code should not compile in any sensible world - I obviously copied extraordinarily wrong.
I was genuinely having an issue, and I never managed to solve it, but I managed to work around it somehow, and didn't have the sense to document what I did. Sorry. Hope I manage to run back into the issue so I can post a proper issue.
